### PR TITLE
[GATE] Authorize user_token and system_token use cases

### PR DESCRIPTION
**Pull request by barbararochazup**
## Issue Description

Endpoint to do authorization by authorization header

Validations:
- Authorization token
- Workspace Access
- Workspace Permissions

Payload:
```json
{
    "path" : "/v2/api-docs"
    "method: "GET"
}
```
### Response:
Success:
> Status: **204 No Content**

Failure:
> Status: **403 Forbidden**
> Status: **500 Internal Server Error**

### CODE REVIEW

File: gate/Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ vendor:
 				$(GOMOD) vendor
 
 mocks:vendor
-				$(GORUN) $(VEND_PATH)/github.com/vektra/mockery/v2/main.go --all --dir ./internal --output ./tests/unit/mocks/ --keeptree --case underscore
+				$(GORUN) $(VEND_PATH)/github.com/vektra/mockery/v2/main.go --all --dir ./internal/repository --output ./tests/unit/mocks/ --keeptree --case underscore
```
Comment (emiteze) [inline gate/Makefile:24]: Why was this path changed?

File: gate/cmd/http.go [hunk 2]
```diff
@@ -116,6 +115,13 @@ func (server server) registerRoutes() {
 				systemToken.GET("", handlers.GetAllSystemTokens(systemTokenInteractor.NewGetAllSystemToken(server.persistenceManager.systemTokenRepository)))
 				systemToken.POST("/:id/revoke", handlers.RevokeSytemToken(systemTokenInteractor.NewRevokeSystemToken(server.persistenceManager.systemTokenRepository)))
 			}
+
+			authorization := v1.Group("/authorization")
```
Comment (emiteze) [inline gate/cmd/http.go:119]: Maybe /authorize would be a better path

File: gate/internal/repository/workspace_repository.go [hunk 3]
```diff
@@ -29,6 +34,41 @@ func (workspaceRepository workspaceRepository) CountByIds(workspaceIds []string)
...
+	return permissionsListDomain, nil
+}
+
 func handleWorkspaceError(message string, operation string, err error, errType string) error {
 	return logging.NewError(message, err, errType, nil, operation)
-}
\ No newline at end of file
+}
+
+const findUserPermissionsAtWorkspace = `
```
Comment (emiteze) [inline gate/internal/repository/workspace_repository.go:67]: findUserPermissionsAtWorkspaceQuery would be a better name

**Review by emiteze — CHANGES_REQUESTED** [review]

File: gate/Makefile (hunk 1 above)
Comment (pedroborgeszup) [reply gate/Makefile:24]: In tests we only use Repository mocks but we were mocking everything, generating unused mocks

**Review by emiteze — APPROVED** [review]

File: gate/internal/repository/models/permission.go [hunk 4]
```diff
@@ -26,5 +26,5 @@ import (
 type Permission struct {
 	ID        uuid.UUID
 	Name      string
-	CreatedAt time.Time
+	CreatedAt *time.Time `json:"-"`
```
Comment (arthurpuccizup) [inline gate/internal/repository/models/permission.go:29]: Why this json description? In theory the model structs will never get above the repository layer, so it's not being transformed to json

File: gate/internal/repository/workspace_repository.go [hunk 5]
```diff
@@ -47,6 +52,41 @@ func (workspaceRepository workspaceRepository) CountByIds(workspaceIds []string)
...
+}
+
+const findUserPermissionsAtWorkspaceQuery = `
+	select wug.permissions
+	from workspaces w
+		inner join workspaces_user_groups wug on wug.workspace_id = ?
+		inner join user_groups ug on ug.id  = wug.user_group_id
+		inner join user_groups_users ugu on ugu.user_group_id = ug.id 
+		inner join users u on ? = ugu.user_id
+`
```
Comment (arthurpuccizup) [inline gate/internal/repository/workspace_repository.go:92]: Query as constant

**Review by monicaribeiro — APPROVED** [review]

**Pull request merged**